services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -239,27 +244,39 @@ def compute_categorical_statistics(
     con: duckdb.DuckDBPyConnection,
     column_name: str,
     table_name: str,
-    class_label_names: list[str],
+    class_label_feature: ClassLabel,
     n_samples: int,
 ) -> CategoricalStatisticsItem:
     categorical_counts_query = COMPUTE_CATEGORIES_COUNTS_COMMAND.format(
         column_name=column_name, data_table_name=table_name
     )
     logging.debug(f"Compute categories counts for {column_name}.\n{categorical_counts_query}")
-    ids_to_counts: dict[int, int] = dict(
+    ids2counts: dict[int, int] = dict(
         con.sql(categorical_counts_query).fetchall()
-    )  # dict {idx: num_samples}; idx might be also None for null values
-    nan_count = ids_to_counts.pop(None, 0)  # type: ignore
-    labels_to_counts: dict[str, int] = {class_label_names[cat_id]: freq for cat_id, freq in ids_to_counts.items()}
-    n_unique = len(labels_to_counts)
+    )  # dict {idx: num_samples}; idx might be also None for null values and -1 for `no label`
+    nan_count = ids2counts.pop(None, 0)  # type: ignore
```

Comment:
to fix the type:ignore issue, maybe dict[Optional[int], int] instead of dict[int, int]

Reply:
oh thank you that worked!